File: sonarqube-companion-frontend/src/app/synchronization/synchronization-state.ts
export interface SynchronizationState {
  startTimestamp: number;
  finishTimestamp: number;
  allTasks: number;
  finishedTasks: number;
  failedTasks: number;
  finished: boolean;
  estimatedDuration: number;
  currentServerTimestamp: number;
  duration: number;
}

File: sonarqube-companion-rest/src/main/java/pl/consdata/ico/sqcompanion/sync/SynchronizationState.java
package pl.consdata.ico.sqcompanion.sync;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SynchronizationState {
    private Long startTimestamp;
    private Long finishTimestamp;
    private Long allTasks;
    private Long finishedTasks;
    private Long failedTasks;
    private Long duration;
    private Double estimatedDuration;
    private boolean finished;
    private Long currentServerTimestamp;
}
